core/templates/components/forms/custom-forms-directives/edit-thumbnail-modal.component.spec.ts

```diff
@@ -144,9 +151,41 @@ describe('Edit Thumbnail Modal Component', () => {
     component.onFileChanged(file);
     expect(component.invalidImageWarningIsShown).toBe(false);
     expect(component.invalidFilenameWarningIsShown).toBe(false);
-    expect(component.tags).toEqual(['script']);
-    expect(component.attrs).toEqual([]);
-    expect(resetSpy).toHaveBeenCalled();
+  });
+
+  it('should remove invalid tags and attributes', ()=>{
+    spyOn(component, 'isUploadedImageSvg').and.returnValue(true);
+    spyOn(component, 'isValidFilename').and.returnValue(true);
+    spyOn(
+      svgSanitizerService, 'getInvalidSvgTagsAndAttrs'
+    ).and.callFake(() => {
+      return { tags: ['circel'], attrs: ['data-name'] };
```

Comment:
I'm confused about the modelling of this data structure. Shouldn't each attr be attached to a tag? An attr name may be valid for one tag but invalid for another.

Reply:
Yes, that was a mistake on my part. Actually, I am working on it and now the method `removeInvalidTagsAndAttributes` looks something like this: 
![Screenshot from 2022-03-01 18-23-27](https://user-images.githubusercontent.com/89375125/156172674-72e0bd1f-4f79-46d7-94df-d15489698cd4.png)

The `attrs` property of invalidTagsAndAttributes parameter is now an array of string values in format `tag:invalid-attribute`. I am updating other files according to the changes. This would also solve code repetitions in various places @seanlip mentioned below where I was parsing `tag:invalid-attribute` to get only the invalid-attribute name.

Reply:
Corrected this one.